.github/workflows/ci.yml

```diff
@@ -19,6 +19,11 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
 
+    - name: Setup Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: '1.22'
```

Comment:
Wouldn't it be better to use https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file ? That way, we test our target version AND we don't have to remember to update the CI code.

Reply:
Yes and no. I was wondering if the best way would be to put the oldest working version in the go.mod so library users are not forced to use the newest Go or actually go for the newest release. What's your point on that?

Reply:
Ah yes, I agree with that. We should keep the Go version inside the go.mod file as lower as possible. But I think this is a different problem than selecting the Go version inside the CI. Anyway, I'm not forcing that, if you're more comfortable with hard-coding the Go version in `ci.yml` :)